test/server/proxy-test.js

```diff
@@ -604,5 +608,25 @@ describe('Proxy', function () {
                 done();
             });
         });
+
+        it('Should recalculate content length after the upload info is injected (GH-195)', function (done) {
+            var src      = newLineReplacer.replace(fs.readFileSync('test/server/data/upload/src.formdata'));
+            var expected = newLineReplacer.replace(fs.readFileSync('test/server/data/upload/expected.formdata'));
+
+            var options = {
+                url:     proxy.openSession('http://127.0.0.1:2000/GH-195/content-length', session),
+                method:  'POST',
+                body:    src,
+                headers: {
+                    'content-type': 'multipart/form-data; boundary=separator',
+                    'accept':       'text/html,application/xhtml+xml,application/xml;q=0.9,*!/!*;q=0.8'
+                }
+            };
+
+            request(options, function (err, res, body) {
+                expect(body.indexOf('%content-length:' + expected.length + '%')).gt(-1);
```

Comment:
Or even better we can send back modified body, and compare it to the `expected`.